src/NuGet.Core/NuGet.Configuration/Settings/SettingFactory.cs

```diff
@@ -99,8 +100,24 @@ internal static SettingBase Parse(XNode node, SettingsFile origin)
         internal static IEnumerable<T> ParseChildren<T>(XElement xElement, SettingsFile origin, bool canBeCleared) where T : SettingElement
         {
             var children = new List<T>();
-
-            var descendants = xElement.Elements().Select(d => Parse(d, origin)).OfType<T>().Distinct();
+            IEnumerable<T> descendants;
+            if (xElement.Name.LocalName.Equals(ConfigurationConstants.PackageSourceMapping, StringComparison.OrdinalIgnoreCase))
+            {
+                descendants = xElement.Elements().Select(d => Parse(d, origin)).OfType<T>();
+                var duplicatedPackageSource = descendants.Where(node => node.ElementName.Equals(ConfigurationConstants.PackageSourceAttribute, StringComparison.OrdinalIgnoreCase))
+                                            .ToLookup(d => d.Attributes["key"], d => d, StringComparer.OrdinalIgnoreCase)
+                                            .Where(g => g.Count() > 1)
+                                            .ToList();
+                if (duplicatedPackageSource.Any())
+                {
+                    var duplicatedKey = string.Join(", ", duplicatedPackageSource.Select(d => d.Key));
```

Comment:
Given that you have origin, you should be able to provide the paths of the duplicated configs no?